crates/uv-python/src/implementation.rs

```diff
@@ -25,8 +25,12 @@ pub enum LenientImplementationName {
 }
 
 impl ImplementationName {
-    pub(crate) fn possible_names() -> impl Iterator<Item = &'static str> {
-        ["cpython", "pypy", "graalpy", "cp", "pp", "gp"].into_iter()
+    pub(crate) fn short_names() -> impl Iterator<Item = &'static str> {
+        ["cp", "pp", "gp"].into_iter()
+    }
+
+    pub(crate) fn long_names() -> impl Iterator<Item = &'static str> {
+        ["cpython", "pypy", "graalpy"].into_iter()
```

Comment:
What's the motivation for splitting these up? (Is it functional?)

Reply:
Oh I see, you use `long_names` elsewhere (per PR summary, to find `pypy`).

Reply:
Yeah, e.g., we don't want to look for interpreters with executables named `cp` but `cp` is a valid request name when parsing so I had to separate them.